echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/artifacts/ArtifactMatcher.java

```diff
@@ -32,10 +33,9 @@
 public class ArtifactMatcher {
   public static Boolean anyArtifactsMatchExpected(List<Artifact> messageArtifacts, Trigger trigger, Pipeline pipeline) {
     List<String> expectedArtifactIds = trigger.getExpectedArtifactIds();
-    List<ExpectedArtifact> expectedArtifacts = pipeline.getExpectedArtifacts()
-        .stream()
-        .filter(e -> expectedArtifactIds.contains(e.getId()))
-        .collect(Collectors.toList());
+    List<ExpectedArtifact> expectedArtifacts = pipeline.getExpectedArtifacts() == null ? new ArrayList<>() : pipeline.getExpectedArtifacts();
```

Comment:
While testing this we ran into a weird issue where a pipeline was returning null for expectedArtifacts so implemented a naive way around null since wasn't too sure what would be the expected response in that case.